src/Sparrow.Server/Compression/AdaptiveMemoryEncoderState.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Buffers;
+
+namespace Sparrow.Server.Compression
+{
+    public unsafe struct AdaptiveMemoryEncoderState : IEncoderState
+    {
+        private byte[] _encodingBuffer;
+        private byte[] _decodingBuffer;
+        private int _size;
+
+        public AdaptiveMemoryEncoderState(int size)
+        {
+            _encodingBuffer = ArrayPool<byte>.Shared.Rent(size);
+            _decodingBuffer = ArrayPool<byte>.Shared.Rent(size);
+            _size = size;
+        }
+
+        public Span<byte> EncodingTable => _encodingBuffer.AsSpan(0, _size);
+        public Span<byte> DecodingTable => _decodingBuffer.AsSpan(0, _size);
+
+        public bool CanGrow => true;
+
+        public void Grow(int minimumSize)
+        {
+            if (_encodingBuffer.Length < minimumSize)
+            {
+                ArrayPool<byte>.Shared.Return(_encodingBuffer);
+                _encodingBuffer = ArrayPool<byte>.Shared.Rent(minimumSize);
```

Comment:
Shouldn't you _copy_ the state first?

Reply:
There is no state when that happens... we know before the fact how big it has to be. Will add some extra documentation to that.

Reply:
Did you add the "extra documentation" you mentioned here?